platform/core/src/services/HangingProtocolService/HangingProtocolService.js

```diff
@@ -339,9 +351,19 @@ class HangingProtocolService {
     // Note: this is happening before displaySets are created. Here, displaySet
     // only contains the information of the id of the displaySet to be matched
     // based on some rules
+    const matchedUids = {};
+    this.customAttributeRetrievalCallbacks.alreadyMatched = (metadata) => {
+      return !!matchedUids[metadata.uid];
+    }
+
     stageModel.displaySets.forEach(displaySet => {
-      const { bestMatch } = this._matchImages(displaySet);
+      const { bestMatch, matchingScores } = this._matchImages(displaySet);
       this.displaySetMatchDetails.set(displaySet.id, bestMatch);
+      if (bestMatch) {
+        const { displaySetInstanceUID } = bestMatch;
+        matchedUids[displaySetInstanceUID] = true;
+        bestMatch.matchingScores = matchingScores;
```

Comment:
The bestMatch also has the list of other matches, in case the viewport species an alternative view.